### DISCUSSION TITLE
Advanced integration specific configuration (From issue #381)

### DISCUSSION THREAD

**Discussion by teharris1** [home-assistant/architecture#632] [General] [answered]
Continuing from issue #381 

Perhaps this issue can be broken down into two subtopics:

- Network / component level configuration which, in theory, can be handled via Config Flow.
- Device specific configuration which does not have a meaningful solution today.

It seems to me it would make sense to have a Config Flow concept for devices. The main advantage of this concept would be to enable device specific configuration without frontend development. The question is how extensible this would need to be for all of the different component specific devices. My gut tells me you have a few types of device configuration concepts:

1. Individual settings (i.e. max brightness, text to display on a device screen, timeout value, enable/disable etc.)
2. Grouped settings (thermostat built in schedule, database of connected devices, lock codes etc.)
3. Actions (refresh device settings, factory reset device, etc.)

Of course there may be more but all of those concepts feel like you could create a config flow function that would appear on the device details page to handle device configuration settings.

There is also a question of how useful Config Flow is for network / component level configuration. MQTT, Z-Wave (OZW, Zwave, Zwave-JS) and ZHA all have custom network level configuration panels built into the front end. I am not sure if this was because of legacy reasons or if the Config Flow features have gaps to allow the work to be done in the Config Flow options only. @balloob I think you are intimately familiar with all of the components with custom panels. Do you believe you could replace the network level configuration with Config Flow options?

**Comment by MartinHjelmare** [comment, marked as answer]
Please continue discussing in https://github.com/home-assistant/architecture/discussions/633 instead.
